app/components/scrollingCalendar/helpers.ts

```diff
@@ -547,3 +555,260 @@ export const tabbingAndArrowNavigation = (
+    if (
+      assignment.startsOn &&
+      assignment.endsOn &&
+      assignment.estimatedWeeklyHours
+    ) {
+      const weeksBetween = getWeeksBetweenDates(
+        assignment.startsOn,
+        assignment.endsOn
+      );
+
+      weeksBetween.forEach((week) => {
+        const key = `${week.year}-${week.cweek}`;
+
+        const isWeekInWorkWeeks = assignment.workWeeks.some(
+          (w) =>
+            w.cweek === week.cweek && w.year === week.year && w.estimatedHours
+        );
+
+        if (!isWeekInWorkWeeks) {
+          if (!weeklyHoursMap[key]) {
+            weeklyHoursMap[key] = {
+              week: week.cweek,
+              year: week.year,
+              totalActualHours: 0,
+              totalEstimatedHours: 0,
+            };
+          }
+
+          weeklyHoursMap[key].totalEstimatedHours +=
+            assignment.estimatedWeeklyHours;
+        }
+      });
+    }
+    assignment.workWeeks.forEach((week) => {
+      const key = `${week.year}-${week.cweek}`;
+
+      if (!weeklyHoursMap[key]) {
+        weeklyHoursMap[key] = {
+          week: week.cweek,
+          year: week.year,
+          totalActualHours: 0,
+          totalEstimatedHours: 0,
+        };
+      }
+
+      weeklyHoursMap[key].totalActualHours += week.actualHours || 0;
+      weeklyHoursMap[key].totalEstimatedHours += week.estimatedHours || 0;
+    });
+  });
+
+  return Object.values(weeklyHoursMap);
+};
+
+export const calculateWeeklyHoursPerProjectForCSV = (
+  weeklyHours: WeeklyHoursSummary[]
+) => {
+  return weeklyHours.reduce(
+    (acc, week) => {
+      acc.totalEstimatedHoursPerProject += week.totalEstimatedHours || 0;
+      acc.totalActualHoursPerProject += week.totalActualHours || 0;
+      return acc;
+    },
+    { totalEstimatedHoursPerProject: 0, totalActualHoursPerProject: 0 }
+  );
+};
+
+export const groupAndSumWeeksByMonthForUsers = (
+  assignments: AssignmentType[]
+) => {
+  const usersByMonth: {
+    userId: string | number;
+    userName: string;
+    months: {
+      monthLabel: string;
+      monthNumber: number;
+      year: number;
+      totalActualHours: number;
+      totalEstimatedHours: number;
+    }[];
+  }[] = [];
+
+  assignments.forEach((assignment) => {
+    const userId = assignment?.assignedUser?.id || "";
+    const userName = assignment?.assignedUser?.name || "TBD user";
+    let userGroup = usersByMonth.find((user) => user.userId === userId);
+    if (!userGroup) {
+      userGroup = {
+        userId,
+        userName,
+        months: [],
+      };
+      usersByMonth.push(userGroup);
+    }
+
+    assignment.workWeeks.forEach((week) => {
+      const startOfWeek = DateTime.fromObject({ year: week.year })
+        .plus({ weeks: week.cweek - 1 })
+        .startOf("week");
+      const month = startOfWeek.monthLong;
+      const monthNumber = startOfWeek.month;
+      const year = startOfWeek.year;
+      const key = `${month} ${year}`;
+
+      let monthGroup = userGroup.months.find(
```

Comment:
this is causing vercel to not deploy because userGroup is potentially undefined

Reply:
@HankC138 thx, will check